superset-frontend/src/dashboard/components/nativeFilters/FilterBar/Vertical.tsx

```diff
@@ -0,0 +1,298 @@
+  width: ${({ theme }) => theme.gridUnit * 8}px;
+
+  & .ant-tabs-top > .ant-tabs-nav {
+    margin: 0;
+  }
+  &.open {
+    width: ${({ width }) => width}px; // arbitrary...
+  }
+`;
+
+const Bar = styled.div<{ width: number }>`
+  & .ant-typography-edit-content {
+    left: 0;
+    margin-top: 0;
+    width: 100%;
+  }
+  position: absolute;
+  top: 0;
+  left: 0;
+  flex-direction: column;
+  flex-grow: 1;
+  width: ${({ width }) => width}px;
+  background: ${({ theme }) => theme.colors.grayscale.light5};
+  border-right: 1px solid ${({ theme }) => theme.colors.grayscale.light2};
+  border-bottom: 1px solid ${({ theme }) => theme.colors.grayscale.light2};
+  min-height: 100%;
+  display: none;
+  &.open {
+    display: flex;
+  }
+`;
+
+const CollapsedBar = styled.div<{ offset: number }>`
+  position: absolute;
+  top: ${({ offset }) => offset}px;
+  left: 0;
+  height: 100%;
+  width: ${({ theme }) => theme.gridUnit * 8}px;
+  padding-top: ${({ theme }) => theme.gridUnit * 2}px;
+  display: none;
+  text-align: center;
+  &.open {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    padding: ${({ theme }) => theme.gridUnit * 2}px;
+  }
+  svg {
+    cursor: pointer;
+  }
+`;
+
+const StyledCollapseIcon = styled(Icons.Collapse)`
+  color: ${({ theme }) => theme.colors.primary.base};
+  margin-bottom: ${({ theme }) => theme.gridUnit * 3}px;
+`;
+
+const StyledFilterIcon = styled(Icons.Filter)`
+  color: ${({ theme }) => theme.colors.grayscale.base};
+`;
+
+const StyledTabs = styled(AntdTabs)`
+  & .ant-tabs-nav-list {
+    width: 100%;
+  }
+  & .ant-tabs-tab {
+    display: flex;
+    justify-content: center;
+    margin: 0;
+    flex: 1;
+  }
+
+  & > .ant-tabs-nav .ant-tabs-nav-operations {
+    display: none;
+  }
+`;
+
+const FilterBarEmptyStateContainer = styled.div`
+  margin-top: ${({ theme }) => theme.gridUnit * 8}px;
+`;
+
+export const FilterBarScrollContext = createContext(false);
+const VerticalFilterBar: React.FC<VerticalBarProps> = ({
+  actions,
+  canEdit,
+  dataMaskSelected,
+  directPathToChild,
+  filtersOpen,
+  filterValues,
+  height,
+  isDisabled,
+  isInitialized,
+  offset,
+  onSelectionChange,
+  toggleFiltersBar,
+  width,
+}) => {
+  const [editFilterSetId, setEditFilterSetId] = useState<number | null>(null);
+  const filterSets = useFilterSets();
+  const filterSetFilterValues = Object.values(filterSets);
+  const [tab, setTab] = useState(TabIds.AllFilters);
+  const nativeFilterValues = filterValues.filter(isNativeFilter);
+  const [isScrolling, setIsScrolling] = useState(false);
+  const timeout = useRef<any>();
+
+  const openFiltersBar = useCallback(
+    () => toggleFiltersBar(true),
+    [toggleFiltersBar],
+  );
+
+  const onScroll = useCallback(
+    throttle(() => {
+      clearTimeout(timeout.current);
+      setIsScrolling(true);
+      timeout.current = setTimeout(() => {
+        setIsScrolling(false);
+      }, 300);
+    }, 200),
+    [],
+  );
```

Comment:
This is functionally equivalent without the warning.

```suggestion
  const onScroll = useMemo(
    () =>
      throttle(() => {
        clearTimeout(timeout.current);
        setIsScrolling(true);
        timeout.current = setTimeout(() => {
          setIsScrolling(false);
        }, 300);
      }, 200),
    [],
  );
```